### PR TITLE
Add job to validate commit messages

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

File: scripts/jenkins/gitlint.ini [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+regex=^SCRD-.*|.*\((bsc#[1-9]+\s?)+\)$
+
+[body-max-line-length]
+line-length=80
+
+[body-min-length]
+min-length=5
+
+[author-valid-email]
+regex = "[^@]+@suse.com"
```
Comment (aspiers) [inline scripts/jenkins/gitlint.ini]: This doesn't look wide enough - you'll need to accommodate at least `@suse.de` too, and maybe others.

**Comment by dirkmueller** [comment]
@rsalevsky so why exactly do we need to enforce SCRD or bsc's in the git commit subject?

**Comment by bmwiedemann** [comment]
@dirkmueller maybe because the subject automatically ends up in rpm .changes files (via trackupstream) which automatically triggers bugzilla updates by my 'obsbugzilla' scripts.
Also other tracking by QAM might be helped.

I guess, as an alternative, we could extend the change-entry generation behind trackupstream to also pull in bug references outside the subject. @aspiers might be interested

**Comment by dirkmueller** [comment]
@bmwiedemann I am okay with referencing bugs if there is a bug referenced. but requiring a bugreport for every commit seems strange

**Comment by rsalevsky** [comment]
I took the current ardana policy as a reference. If I'm not mistaken then it is a SCRD or a bsc reference correct?

**Comment by bmwiedemann** [comment]
Other projects solve this by marking certain patches as "trivial" that dont require NEWS entries and bug refs.
E.g. we dont want to open a bug for every documentation typo fix. Probably users wouldnt even want to read it in the changelog, because it makes finding relevant changes harder (aka SNR)

File: scripts/jenkins/gitlint.ini [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+verbosity = 2
+
+[title-max-length]
+line-length=80
+
+[title-must-not-contain-word]
+words=wip,WIP
+
+[title-match-regex]
+regex=^SCRD-.*|.*\((bsc#[1-9]+\s?)+\)$
```
Comment (aojea) [inline scripts/jenkins/gitlint.ini]: I think that we can add trello cards id too

Comment (rsalevsky) [reply scripts/jenkins/gitlint.ini]: How does a trello card id look like?

Comment (jesusaurus) [reply scripts/jenkins/gitlint.ini]: The trello card id is an alphanumeric string, you can find a sed script that turns references to bugs into links to the bugs at http://git.suse.provo.cloud/cgit/hp/config/tree/modules/hp/files/git/commit-filter.sh?h=suse/gozer for reference.

**Comment by rsalevsky** [comment]
@jesusaurus I don't think there is much to discuss as there is a technical reason for this. For the package changelog we only use the header and as the bsc has to be visible in the changelog it has to be in the header.

**Comment by aspiers** [comment]
[**@bmwiedemann**](https://github.com/bmwiedemann) commented on [10 Sep 2018, 09:05 BST](https://github.com/SUSE-Cloud/automation/pull/2720#issuecomment-419824329 "2018-09-10T08:05:59Z - Replied by Github Reply Comments"):
> @dirkmueller maybe because the subject automatically ends up in rpm .changes files (via trackupstream) which automatically triggers bugzilla updates by my 'obsbugzilla' scripts.
> Also other tracking by QAM might be helped.
> 
> I guess, as an alternative, we could extend the change-entry generation behind trackupstream to also pull in bug references outside the subject. @aspiers might be interested

@bmwiedemann I proposed this ~11 months ago: https://github.com/openSUSE/obs-service-tar_scm/issues/207

If you are in favour then please add your support there and help the objectors to see the light ;-)

**Comment by aspiers** [comment]
[**@rsalevsky**](https://github.com/rsalevsky) commented on [18 Sep 2018, 15:39 BST](https://github.com/SUSE-Cloud/automation/pull/2720#issuecomment-422420378 "2018-09-18T14:39:18Z - Replied by Github Reply Comments"):
>@jesusaurus wrote:
>>The job looks good as-is, but this change should also add a trigger so that the job will run automatically.
>>
>>Also, this seems to be a good time to re-consider requiring bugs/tickets in the commit subject (I personally think they belong in the commit footer, but lost that battle last time around);

@jesusaurus You didn't lose that battle because it hasn't even really started in earnest yet, let alone been won or lost :-)

>>though that is a discussion is too large to happen in review comments, so maybe it's time to start an email thread on the topic?

I already made the proposal in https://github.com/openSUSE/obs-service-tar_scm/issues/207 - please add your feedback there.

> @jesusaurus I don't think there is much to discuss as there is a technical reason for this. For the package changelog we only use the header and as the bsc has to be visible in the changelog it has to be in the header.

@rsalevsky You are assuming the status quo cannot change, but that is not true.  As [I wrote about this topic on `cloud-devel` last year](http://mailman.suse.de/mlarch/SuSE/cloud-devel/2017/cloud-devel.2017.08/msg00202.html):

>Let's ensure the tools work for us, rather than us working for the tools ;-)

**Comment by aojea** [comment]
@jesusaurus please go ahead, I think that at the beginning it slows thing but in the mid/long term it forces us to follow best practices and document better the changes

> Also, this seems to be a good time to re-consider requiring bugs/tickets in the commit subject (I personally think they belong in the commit footer, but lost that battle last time around); though that is a discussion is too large to happen in review comments, so maybe it's time to start an email thread on the topic?

**Comment by aspiers** [comment]
[**@aojea**](https://github.com/aojea) commented on [19 Sep 2018, 08:52 BST](https://github.com/SUSE-Cloud/automation/pull/2720#issuecomment-422696485 "2018-09-19T07:52:20Z - Replied by Github Reply Comments"):
> @jesusaurus please go ahead, I think that at the beginning it slows thing but in the mid/long term it forces us to follow best practices and document better the changes

I agree it's a discussion worth having sooner rather than later.

> > Also, this seems to be a good time to re-consider requiring bugs/tickets in the commit subject (I personally think they belong in the commit footer, but lost that battle last time around); though that is a discussion is too large to happen in review comments, so maybe it's time to start an email thread on the topic?

Or maybe just propose a spec and announce it on cloud-devel, so that the majority of discussion happens as a review on the spec?  I suspect we'll make faster progress that way.  It's too easy for a mailing list thread to turn into bike-shedding ;-)

**Pull request merged**